[RFC] UI fine tuning for the dive video feature

sfuchs79 opened the pull request:
<!-- Lines like this one are comments and will not be shown in the final output. -->
<!-- Make sure that you have read the "Contributing" section of the README and also the notes in CodingStyle. -->
<!-- If you are a collaborator, please add labels and assign other collaborators for a review. -->

### Describe the pull request:
<!-- Replace [ ] with [x] to select options. -->
- [ ] Bug fix
- [x] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [x] Language translation

### Pull request long description:
<!-- Describe your pull request in detail. -->
1) UI change of "images"/"photo" to "media" or "media files"

This changes the above mentioned terms everywhere in the UI to
reflect the fact that Subsurface now also supports video files on top
of image files.

2) For media file open dialog add different file filters

On top of the file filter for all media files add a file filter
for images only and one for videos only.

### Changes made:
<!-- Enumerate the changes with 1), 2), 3) etc. -->
<!-- Ensure the test cases are updated if needed. -->

### Related issues:
<!-- Reference issues with #<issue-num>. -->
<!-- Write "Fixes #<issue-num" to notify Github that this PR fixes an issue. -->

### Additional information:
<!-- Include sample dive log or other relevant information to allow testing the change where feasible. -->
I'm 100% sure that I missed some strings. Would be great if s.o. could search the UI for missing changes from "images" to "media".


### Release note:
<!-- Describe if this change needs a release note present in CHANGELOG.md. -->
<!-- Also, please make sure to add the release note on top of the file CHANGELOG.md. -->
Yes, will follow after we decide if and how we're going to change this.

### Documentation change:
<!-- If this PR makes changes to user functionality, then the documentation has to be updated too. -->
<!-- Please, briefly outline here what has changed in terms of the user experience (UX). -->
<!-- If UX changes have been made, a maintainer should apply the 'needs-documentation-change' label. -->

### Mentions:
<!-- Mention users that you want to review your pull request with @<user-name>. Leave empty if not sure. -->
@bstoeger @atdotde @neolit123

bstoeger commented on desktop-widgets/divelistview.cpp:
```diff
@@ -922,11 +922,14 @@ void DiveListView::shiftTimes()
 							      lastUsedImageDir(),
-							      tr("Image files (%1)").arg(filters.join(" ")));
+							      tr("Media files (%1);;Image files (%2);;Video files (%3)")
+							      .arg(m_filters.join(" ")).arg(i_filters.join(" ")).arg(v_filters.join(" ")));
```
If all arguments are strings, I think the more efficient (and safer) method is:
```
x.arg(string1, string2, string3)
```
The reason is: suppose string1 contains "%1" or something, then
```
x.arg(string1).arg(string2).arg(string3)
```
will do funky things.

neolit123 commented:
> Derek add label: post-4.8.1
Derek add label: dive-video
Derek add label: rfc
Derek add label: translation

the bot isn't working, let me try.

neolit123 commented:
Derek add label: dive-video

neolit123 commented:
@sfuchs79 no idea why you are part of `maintainers.

bstoeger commented:
> Derek add label: post-4.8.1

My impatient side would prefer to have this in post-4.8.1, but you're probably right.

sfuchs79 commented:
@bstoeger 
Thanks for the review and feedback!!! I'll take care of all the items.
Sorry, but your impatient side needs to be a little bit more patient because I'll go diving tomorrow and therefore I have to go to bed now ;-)
Plus we should wait for some more feedback regarding the overall change from "images" to "media". Plus the one commit is useless w/o the translators translating the new strings.
:-)

@neolit123 
The bot hates me?!?

sfuchs79 commented:
@bstoeger 
> Could we also have an "All files" option in the file dialog? In case we don't guess all video extensions. It annoyed me during testing that it won't show all files.

Sure, good idea. "All files" at the very bottom of the list?

bstoeger commented:
> "All files" at the very bottom of the list?

Yes, I think that's how it is usually done.

Happy diving. :)

dirkhh commented:
I believe the bot understands only one command per comment.
@sfuchs79 when you are back from diving, give that a try :-) And yes, definitely this is post-4.8.1

sfuchs79 commented:
Derek add label: translation

dirkhh commented:
Derek add label: translation

dirkhh commented:
Hey, Derek, you're a bit of a jerk not listening to @sfuchs79 ...

sfuchs79 commented:
@dirkhh 
You may remember: I once complained that Dereks avatar looks pretty evil. May he/she/it is still upset.

dirkhh commented:
That seems reasonable, but hey @alexellis - any idea why @sfuchs79 can't use Derek?
He's in the .DEREK.yml file :-)

alexellis commented:
Hi, it's Alex - Derek maintainer. Commands have to be specified one per comment. There's also a 5 minute gap between editing the .DEREK.yml file and the GitHub CDN refreshing. 

@sfuchs79 if you can send another message then I'll check the logs. There is a lot of noise in the logs from the Moby project. 

Alex

alexellis commented:
When Dirk ran the command:

```
derek-054.1.9rk12e33gpm4@faas-combined    | 2018/07/14 15:07:30 Subsurface-divelog Subsurface-divelog [alexellis moby openfaas rgee0 openfaas-incubator Subsurface-divelog]
derek-054.1.9rk12e33gpm4@faas-combined    | dirkhh wants to add label of 'translation' on issue #1507 
derek-054.1.9rk12e33gpm4@faas-combined    | Request to add label of 'translation' on issue #1507 was successfully completed.2018/07/14 15:07:31 Subsurface-divelog Subsurface-divelog [alexellis moby openfaas rgee0 openfaas-incubator Subsurface-divelog]
```

When @sfuchs79 ran the command:

First attempt (too many commands in one)

```
derek-054.1.9rk12e33gpm4@faas-combined    | 2018/07/13 19:28:21 Subsurface-divelog Subsurface-divelog [alexellis moby openfaas rgee0 openfaas-incubator Subsurface-divelog]
derek-054.1.9rk12e33gpm4@faas-combined    | sfuchs79 wants to add label of 'post-4.8.1
derek-054.1.9rk12e33gpm4@faas-combined    | Derek add label: dive-video
derek-054.1.9rk12e33gpm4@faas-combined    | Derek add label: rfc
derek-054.1.9rk12e33gpm4@faas-combined    | Derek add label: translation' on issue #1507 
derek-054.1.9rk12e33gpm4@faas-combined    | POST https://api.github.com/repos/Subsurface-divelog/subsurface/issues/1507/labels: 422 Validation Failed [{Resource:Label Field:name Code:invalid Message:}]
```

Second attempt - I can't find this in the logs. I'd just say make sure there is no new-line added and ask you to try again. (Editing a comment isn't support either)

Alex

sfuchs79 commented:
Derek add label: rfc

alexellis commented:
LGTM @sfuchs79 - let us know over on alexellis/derek if there are any other issues in the future.

Alex

sfuchs79 commented:
@alexellis 
Thanks!

@bstoeger 
- Addressed all your items (hopefully correctly)
- Added a "All files (\*.\*)" filter
- Regarding sorting of file extensions: I checked and we currently also do NOT sort file extensions alphabetically in e.g. dive log file open dialog
- Changed to a better way how we give the file filter strings to Transifex (less complex strings)
- Added a CHANGELOG.md entry

bstoeger commented:
Tried it and works for me. From my side this can be committed, but let's perhaps wait for a short while for opinions on the new strings.

BTW: Since one of the recent commits the fonts got significantly bigger and it spills a strange message on the console:
```
QFont::setPointSizeF: Point size <= 0 (-1.000000), must be greater than 0
```

Huh?

janiversen commented:
I might have a clue, will look tomorrow

janiversen commented:
may I suggest that we pass it by the translators before deciding, I just learned today that “media” as a term for combining e.g. pictures video etc does not trnslate well to spanish.

This is not to stop anything, but just to raise a flag of concern (which might not be correct since I am noway a native soanish or english speaker)

sfuchs79 commented:
I wrote a mail on the mailing list with a link to this PR.
Then I think we should wait a little bit but maybe also not too long. W.c. what can happen is that a translator suggests a change in the source string(s) exactly at the moment when he/she sees it on Transifex - after it had been merged. But then it's still not too late to adapt it again.

dirkhh commented:
Yes. Well done. I really like it.
